plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -345,4 +348,44 @@ export class HogTransformerService {
             : this.hogExecutor.execute(invocation, { functions: transformationFunctions })
         return result
     }
+
+    private async handleCacheMiss(hogFunctionId: string): Promise<HogWatcherState | null> {
+        logger.error('⚠️', 'HogFunction state cache miss - this should not happen', { hogFunctionId })
```

Comment:
If this should genuinely never happen... Shouldn't we just crash out? Thinking about how confusing it would be if we suddenly saw perf issues in ingestion and eventually found this log indicating something was wrong long after the breaking commit is merged.